### PR TITLE
[pytorch] Fix torch.nn.functional.normalize to be properly scriptable

### PR DESCRIPTION
**Pull request by esqu1**
Summary:
Several scenarios don't work when trying to script `F.normalize`, notably when you try to symbolically trace through it with using the default argument:

```
import torch.nn.functional as F
import torch
from torch.fx import symbolic_trace

def f(x):
    return F.normalize(x)

gm = symbolic_trace(f)
torch.jit.script(gm)
```
which leads to the error
```
RuntimeError:

normalize(Tensor input, float p=2., int dim=1, float eps=9.9999999999999998e-13, Tensor? out=None) -> (Tensor):
Expected a value of type 'float' for argument 'p' but instead found type 'int'.
:
def forward(self, x):
    normalize_1 = torch.nn.functional.normalize(x, p = 2, dim = 1, eps = 1e-12, out = None);  x = None
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    return normalize_1

Differential Revision: D26324308

### CODE REVIEW

**Review by jamesr66a — APPROVED** [review]
This looks good to me.

cc @szagoruyko who added the code initially
cc @gmagogsfm in case there are any BC concerns about this. Not sure if this default would have been baked in anywhere in TorchScript but worth looking into it

**Comment by gmagogsfm** [comment]
> This looks good to me.
> 
> cc @szagoruyko who added the code initially
> cc @gmagogsfm in case there are any BC concerns about this. Not sure if this default would have been baked in anywhere in TorchScript but worth looking into it

I am not aware of any such "baking in". As long as all test pass, I am comfortable landing it.
